redfish_utilities/logs.py

```diff
@@ -0,0 +1,181 @@
+#! /usr/bin/python
+# Copyright Notice:
+# Copyright 2020 DMTF. All rights reserved.
+# License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/Redfish-Tacklebox/blob/master/LICENSE.md
+
+"""
+Logs Module
+
+File : logs.py
+
+Brief : This file contains the definitions and functionalities for interacting
+        with the log service for a given Redfish service
+"""
+
+import os
+
+from .messages import verify_response
+from enum import Enum
+
+class RedfishLogServiceNotFoundError( Exception ):
+    """
+    Raised when a matching log service cannot be found
+    """
+    pass
+
+class RedfishClearLogNotFoundError( Exception ):
+    """
+    Raised when a log service does not contain the clear log action
+    """
+    pass
+
+class log_container( Enum ) :
+    """
+    Types of resources that contain log services
+    """
+    MANAGER = "Managers"
+    CHASSIS = "Chassis"
+    SYSTEM = "Systems"
+
+def get_log_entries( context, container_type = log_container.MANAGER, container_id = None, log_service_id = None ):
+    """
+    Finds the log entries of a log service matching the given ID
+
+    Args:
+        context: The Redfish client object with an open session
+        container_type: The type of resource containing the log service (manager, system, or chassis)
+        container_id: The container instance with the log service; if None, perform on the only container
+        log_service_id: The log service with the logs; if None, perform on the only log service
+
+    Returns:
+        An array of log entries
+    """
+
+    log_service = get_log_service( context, container_type, container_id, log_service_id )
+
+    # Read in the log entries
+    log_entries = []
+    log_entry_col = context.get( log_service.dict["Entries"]["@odata.id"] )
+    log_entries.extend( log_entry_col.dict["Members"] )
+
+    # If a next link is provided, iterate over it and add to the log entry list
+    while "Members@odata.nextLink" in log_entry_col.dict:
+        log_entry_col = context.get( log_entry_col.dict["Members@odata.nextLink"] )
+        log_entries.extend( log_entry_col.dict["Members"] )
+
+    return log_entries
+
+def print_log_entries( log_entries, details = False ):
+    """
+    Prints a set of log entries in a table
+
+    Args:
+        log_entries: The log entries to print
+        details: Flag indicating if details should be displayed
+    """
+
+    # Set up templates
+    console_size = os.get_terminal_size()
+    message_size = console_size.columns - 38
+    entry_line_format = "  {:5s} | {:25s} | {}"
+    detail_line_format = "  {:33s} | {}: {}"
```

Comment:
I've been trying to think of a cleaner way of doing this, but have been unsuccessful so far. Essentially, if we ever tweak one of the numbers in the above three lines, the other two lines need a similar adjustment. It works as is, but I would like to think of something more future proof.